pkg/acsengine/pki.go

```diff
@@ -87,22 +92,74 @@ func CreatePki(extraFQDNs []string, extraIPs []net.IP, clusterDomain string, caP
 		errors <- err
 	}()
 
+	go func() {
+		var err error
+		organization := make([]string, 1)
+		organization[0] = "system:masters"
+		ip := net.ParseIP("127.0.0.1").To4()
+		peerIPs := append(extraIPs, ip)
+		etcdServerCertificate, etcdServerPrivateKey, err = createCertificate("etcd", caCertificate, caPrivateKey, false, nil, peerIPs, organization)
+		errors <- err
+	}()
+
+	go func() {
+		var err error
+		organization := make([]string, 1)
+		organization[0] = "system:masters"
+		ip := net.ParseIP("127.0.0.1").To4()
+		peerIPs := append(extraIPs, ip)
+		etcdClientCertificate, etcdClientPrivateKey, err = createCertificate("etcd", caCertificate, caPrivateKey, false, nil, peerIPs, organization)
+		errors <- err
+	}()
+
+	for i := 0; i < 5; i++ {
+		go func() {
+			var err error
+			organization := make([]string, 1)
+			organization[0] = "system:masters"
+			ip := net.ParseIP("127.0.0.1").To4()
+			peerIPs := append(extraIPs, ip)
+			etcdPeerCertificate := new(x509.Certificate)
+			etcdPeerPrivateKey := new(rsa.PrivateKey)
+			etcdPeerCertificate, etcdPeerPrivateKey, err = createCertificate("etcd", caCertificate, caPrivateKey, false, nil, peerIPs, organization)
+			etcdPeerCertPairs = append(etcdPeerCertPairs, &PkiKeyCertPair{CertificatePem: string(certificateToPem(etcdPeerCertificate.Raw)), PrivateKeyPem: string(privateKeyToPem(etcdPeerPrivateKey))})
+			errors <- err
+		}()
+	}
+
 	e1 := <-errors
 	e2 := <-errors
 	e3 := <-errors
+	e4 := <-errors
+	e5 := <-errors
+	e6 := <-errors
 	if e1 != nil {
-		return nil, nil, nil, e1
+		return nil, nil, nil, nil, nil, nil, e1
 	}
 	if e2 != nil {
-		return nil, nil, nil, e2
+		return nil, nil, nil, nil, nil, nil, e2
 	}
 	if e3 != nil {
-		return nil, nil, nil, e2
+		return nil, nil, nil, nil, nil, nil, e3
+	}
+	if e4 != nil {
+		return nil, nil, nil, nil, nil, nil, e4
+	}
+	if e5 != nil {
+		return nil, nil, nil, nil, nil, nil, e5
 	}
+	if e6 != nil {
+		return nil, nil, nil, nil, nil, nil, e5
+	}
+
+	time.Sleep(10 * time.Second)
```

Comment:
I had to add this line here otherwise the go routines didn't have time to finish executing before return and the certs wouldn't get completely generated. Looking for suggestions to improve. cc @anhowe since you're the one who added the go func()'s in this file.